Graphics/Drawers/DrawerManager.cs

```diff
@@ -0,0 +1,186 @@
+
+            foreach (var drawer in drawers)
+                drawer.DrawTarget(Main.spriteBatch);
+
+            Main.spriteBatch.ExitShaderRegion();
+        }
+
+        private void DrawProjectileDrawers(On_Main.orig_DrawProjectiles orig, Main self)
+        {
+            orig(self);
+
+            var drawers = Drawers.Where(drawer => drawer.ShouldDraw && drawer.Layer is DrawerLayer.Projectile);
+
+            Main.spriteBatch.Begin(SpriteSortMode.Immediate, BlendState.AlphaBlend, Main.DefaultSamplerState, DepthStencilState.None, Main.Rasterizer);
+
+            foreach (var drawer in drawers)
+                drawer.DrawTarget(Main.spriteBatch);
+
+            Main.spriteBatch.End();
+        }
+
+        private void DrawPlayerDrawers(On_Main.orig_DrawPlayers_AfterProjectiles orig, Main self)
+        {
+            orig(self);
+
+            var drawers = Drawers.Where(drawer => drawer.ShouldDraw && drawer.Layer is DrawerLayer.Player);
+
+            Main.spriteBatch.Begin(SpriteSortMode.Immediate, BlendState.AlphaBlend, Main.DefaultSamplerState, DepthStencilState.None, Main.Rasterizer);
+
+            foreach (var drawer in drawers)
+                drawer.DrawTarget(Main.spriteBatch);
+
+            Main.spriteBatch.End();
+        }
+
+        private void DrawBeforeTileDrawers(On_Main.orig_DoDraw_WallsTilesNPCs orig, Main self)
+        {
+            var drawers = Drawers.Where(drawer => drawer.ShouldDraw && drawer.Layer is DrawerLayer.BeforeTiles);
+
+            Main.spriteBatch.End();
+            Main.spriteBatch.Begin(SpriteSortMode.Immediate, BlendState.AlphaBlend, Main.DefaultSamplerState, DepthStencilState.None, Main.Rasterizer);
+            foreach (var drawer in drawers)
+                drawer.DrawTarget(Main.spriteBatch);
+
+            Main.spriteBatch.End();
+            Main.spriteBatch.Begin(SpriteSortMode.Deferred, BlendState.AlphaBlend, Main.DefaultSamplerState, DepthStencilState.None, Main.Rasterizer, null, Main.Transform);
+
+            orig(self);
+        }
+
+        private void DrawAfterEverythingDrawers(On_Main.orig_DrawInfernoRings orig, Main self)
+        {
+            var drawers = Drawers.Where(drawer => drawer.ShouldDraw && drawer.Layer is DrawerLayer.AfterEverything);
+
+            Main.spriteBatch.End();
+            Main.spriteBatch.Begin(SpriteSortMode.Immediate, BlendState.AlphaBlend, Main.DefaultSamplerState, DepthStencilState.None, Main.Rasterizer);
+            foreach (var drawer in drawers)
+                drawer.DrawTarget(Main.spriteBatch);
+
+            Main.spriteBatch.End();
+            Main.spriteBatch.Begin(SpriteSortMode.Deferred, BlendState.AlphaBlend, Main.DefaultSamplerState, DepthStencilState.None, Main.Rasterizer, null, Main.Transform);
+            orig(self);
+        }
+        #endregion
+
+        #region Misc Methods
+        public static T GetDrawer<T>() where T : BaseDrawer
```

Comment:
Like above, just move these to `ModType`s as tModLoader intends and expects you to do. Stops you from needing this boilerplate, and it's more efficiently implemented and works with other mods.